core/capabilities/remote/target/client.go

```diff
@@ -121,8 +134,7 @@ func (c *client) Execute(ctx context.Context, capReq commoncap.CapabilityRequest
 		return nil, fmt.Errorf("request for message ID %s already exists", messageID)
 	}
 
```

Comment:
The context below was leaking a goroutine as not being cancelled. As the the requests are now explcitly cancelled by the client on expiry (if not already responded to) there is no need to pass in a context here so the problem goes away.